app/src/main/java/sensors_in_paradise/sonar/JSONStorage.kt

```diff
@@ -12,16 +12,16 @@ abstract class JSONStorage @Throws(
     IOException::class,
     SecurityException::class,
     JSONException::class
-) constructor(val file: File, initialJson: JSONObject? = null) {
-    constructor(file: File) : this(file, null)
+) constructor(val file: File?, initialJson: JSONObject? = null) {
+    constructor(file: File?) : this(file, null)
```

Comment:
I am not sure why an instance of JSONStorage without a file object should be allowed. 

**Most important question in this review**

Reply:
Super!